src/test/librbd/fsx.cc

```diff
@@ -2028,7 +2022,7 @@ main(int argc, char **argv)
 			break;
 		case 'l':
 			maxfilelen = getnum(optarg, &endp);
-			if (maxfilelen <= 0)
+			if (maxfilelen == 0)
```

Comment:
I may should have fixed that in a separate patch, was accidentally mixed into one.

But about this line: maxfilelen is of type 'unsigned long' and can't therefore be negative, that is why checking for less than zero makes absolutely no sense. We may should use a temporary integer variable in this case to take the result of getnum(), check the result and assign than to  maxfilelen if valid. I can change it to this.